irr/include/SSkinMeshBuffer.h

```diff
@@ -301,7 +301,26 @@ struct SSkinMeshBuffer : public IMeshBuffer
 	}
 
 	//! append the vertices and indices to the current buffer
-	void append(const void *const vertices, u32 numVertices, const u16 *const indices, u32 numIndices) override {}
+	void append(const void* const vertices, u32 numVertices, const u16* const indices, u32 numIndices) override {
```

Comment:
btw all variables are public so there's no need to implement this function just to fill the vertex buffer

Reply:
Encapsulation. Just because all variables are public doesn't mean they should be and that we should make the problem worse.

Reply:
It's also frustrating for an API to say "if implemented it will do x, but it may not be implemented, good luck". Which in effect means we cannot call `append` on any supertype without doing a runtime type check to make sure it isn't an `SSkinMeshBuffer` specifically.

Reply:
In fact, without having some `isAppendSupported` method or an exception thrown, there's no way to truly use that API safely whatsoever. The previous implementation that does nothing makes it pointless to have that method in the API in a superclass.

Reply:
an appropriate moving constructor lets us avoid touching this at all.